packages/material-ui/src/Typography/Typography.d.ts

```diff
@@ -1,29 +1,39 @@
 import * as React from 'react';
 import { StandardProps, PropTypes } from '..';
+import { OverrideProps, OverridableTypeMap, OverridableComponent } from '../OverridableComponent';
 import { ThemeStyle } from '../styles/createTypography';
 
 type Style = ThemeStyle | 'srOnly';
 
-export interface TypographyProps
-  extends StandardProps<React.HTMLAttributes<HTMLElement>, TypographyClassKey> {
-  align?: PropTypes.Alignment;
-  color?:
-    | 'initial'
-    | 'inherit'
-    | 'primary'
-    | 'secondary'
-    | 'textPrimary'
-    | 'textSecondary'
-    | 'error';
-  component?: React.ElementType<React.HTMLAttributes<HTMLElement>>;
-  display?: 'initial' | 'block' | 'inline';
-  gutterBottom?: boolean;
-  noWrap?: boolean;
-  paragraph?: boolean;
-  variant?: Style | 'inherit';
-  variantMapping?: Partial<Record<Style, string>>;
+export interface TypographyTypeMap<P = {}, D extends React.ElementType = 'span'> {
+  props: P & {
+    align?: PropTypes.Alignment;
+    color?:
+      | 'initial'
+      | 'inherit'
+      | 'primary'
+      | 'secondary'
+      | 'textPrimary'
+      | 'textSecondary'
+      | 'error';
+    display?: 'initial' | 'block' | 'inline';
+    gutterBottom?: boolean;
+    noWrap?: boolean;
+    paragraph?: boolean;
+    variant?: Style | 'inherit';
+    variantMapping?: Partial<Record<Style, string>>;
+  };
+  defaultComponent: D;
+  classKey: TypographyClassKey;
 }
 
+declare const Typography: OverridableComponent<TypographyTypeMap>;
```

Comment:
This change makes the export of `TypographyProps` unused, does it not @fyodore82?   So code that used to reference and `Pick` from `TypographyProps` I see is now broken and a mismatch for the `Typography` component itself.

Reply:
I see this is an updated pattern elsewhere in the code (e.g. Chip, Grid, Link etc), but unsure how the `*Props` exports relate.

Reply:
Pattern I've employed for Typography is used in several other components in Material-UI as you already mentioned. So I've decided not to invent different solution but follow already established mainline.

The `TypographyProps` and `Typography` itself are related. Both inherit from [`OverrideProps`](https://github.com/mui-org/material-ui/blob/a5a7bdb22e61d5f76d65a79ca665fb23438542a9/packages/material-ui/src/OverridableComponent.d.ts#L11) interface. `Typography` does it indirectly, rhru [`OverridableComponent`](https://github.com/mui-org/material-ui/blob/a5a7bdb22e61d5f76d65a79ca665fb23438542a9/packages/material-ui/src/OverridableComponent.d.ts#L10) and `TypographyProps` does it directly.

`forwardRef` will also work fine if specialize `TypographyProps` with correct component

```
export const Typo = React.forwardRef<any, TypographyProps<'a', {component: 'a'}>>((props, ref) => (
    <Typography variant="body2" ref={ref} {...props} />
))

// now valid
<Typo component="a" href="url" display="block" />
```

I have also checked how it was before `Typography` change. This code `<Typo component="a" href="url" display="block" />` was not valid due to presence of `href` attribute. And we had no means to make it valid.

I may suggest to extend Typography documentation and include samples of how to use component prop in different scenarios

Reply:
@fyodore82 So was the documentation ever updated? Because now I can't use `FC<TypographyProps>` anymore, which was a bit of a low blow to sneak into a minor release.

Reply:
[Did it](https://github.com/mui-org/material-ui/pull/19341)